java/src/main/java/ai/onnxruntime/OnnxSparseTensor.java

```diff
@@ -0,0 +1,920 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package ai.onnxruntime;
+
+import static ai.onnxruntime.OnnxTensor.fp16ToFloat;
+
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.DoubleBuffer;
+import java.nio.FloatBuffer;
+import java.nio.IntBuffer;
+import java.nio.LongBuffer;
+import java.nio.ShortBuffer;
+import java.util.Arrays;
+
+/**
+ * A Java object wrapping an OnnxSparseTensor.
+ *
+ * <p>Sparse tensors support a variety of formats, and the {@link #getValue} method returns a
+ * different static inner class representing each type.
+ */
+public final class OnnxSparseTensor extends OnnxTensorLike {
+  private final SparseTensorType sparseTensorType;
+
+  // Held to prevent deallocation while used in native code.
+  private final Buffer indices;
+  private final LongBuffer innerIndices;
+  private final Buffer values;
+
+  /**
+   * Construct a sparse tensor from JNI.
+   *
+   * @param nativeHandle The tensor native handle.
+   * @param allocatorHandle The allocator handle.
+   * @param sparseType The sparsity type.
+   * @param info The tensor info.
+   */
+  OnnxSparseTensor(long nativeHandle, long allocatorHandle, int sparseType, TensorInfo info) {
+    this(
+        nativeHandle,
+        allocatorHandle,
+        SparseTensorType.mapFromInt(sparseType),
+        info,
+        null,
+        null,
+        null);
+  }
+
+  /**
+   * Construct a COO or block sparse tensor.
```

Comment:
>Construct a COO or block sparse tensor. [](http://example.com/codeflow?start=5&length=39)

I notice that the same code is used with COO and blocksparse tensors. However, it is not clear how they can be compatible.
COO may have either 1-D indices or 2-D. And blocksparse is always 2-D.

Reply:
The factory methods handle that, this is just to get the pointers through into the right place. In both the COO and block sparse cases there is only a single pointer to the indices, which is what this cares about.